daemon/src/model/cfd.rs

```diff
@@ -2272,6 +2290,52 @@ mod tests {
         assert_eq!(funding_fee.rate, funding_rate);
     }
 
+    #[test]
+    fn given_negative_funding_rate_then_long_margin() {
+        let price = Price::new(dec!(10000)).unwrap();
+        let quantity = Usd::new(dec!(200));
+        let leverage = Leverage::new(2).unwrap();
+
+        let funding_rate = dec!(-1);
+
+        let cfd = Cfd::dummy_not_open_yet()
+            .with_position(Position::Long)
+            .with_margin(price, quantity, leverage);
+
+        let margin = cfd.margin_for_funding_rate(FundingRate::new(funding_rate).unwrap());
+        assert_eq!(margin, Amount::from_btc(0.01).unwrap());
+
+        let cfd = Cfd::dummy_not_open_yet()
+            .with_position(Position::Short)
+            .with_margin(price, quantity, leverage);
+
+        let margin = cfd.margin_for_funding_rate(FundingRate::new(funding_rate).unwrap());
+        assert_eq!(margin, Amount::from_btc(0.01).unwrap());
+    }
```

Comment:
Arrange - act- assert :)

```suggestion
    #[test]
    fn given_negative_funding_rate_then_long_margin() {
        let price = Price::new(dec!(10000)).unwrap();
        let quantity = Usd::new(dec!(200));
        let leverage = Leverage::new(2).unwrap();
        let funding_rate = FundingRate::new(dec!(-1)).unwrap();

        let margin_on_long_cfd = Cfd::dummy_not_open_yet()
            .with_position(Position::Long)
            .with_margin(price, quantity, leverage)
            .margin_for_funding_rate(funding_rate)
            .unwrap();
        let margin_on_short_cfd = Cfd::dummy_not_open_yet()
            .with_position(Position::Short)
            .with_margin(price, quantity, leverage)
            .margin_for_funding_rate(funding_rate)
            .unwrap();
            
        assert_eq!(margin_on_long_cfd, Amount::from_btc(0.01).unwrap());
        assert_eq!(margin_on_short_cfd, Amount::from_btc(0.01).unwrap());
    }
```

Reply:
I added a fixup - not exactly what you propose, but you will like it I think :)